lib/model/model.go

```diff
@@ -2538,42 +2558,39 @@ func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly
 
 		f.Name = strings.Replace(f.Name, prefix, "", 1)
 
-		var dir, base string
-		if f.IsDirectory() && !f.IsSymlink() {
-			dir = f.Name
-		} else {
-			dir = filepath.Dir(f.Name)
-			base = filepath.Base(f.Name)
-		}
+		dir := filepath.Dir(f.Name)
+		base := filepath.Base(f.Name)
 
 		if levels > -1 && strings.Count(f.Name, sep) > levels {
 			return true
 		}
 
-		last := output
+		parent := root
 		if dir != "." {
 			for _, path := range strings.Split(dir, sep) {
-				directory, ok := last[path]
-				if !ok {
-					newdir := make(map[string]interface{})
-					last[path] = newdir
-					last = newdir
-				} else {
-					last = directory.(map[string]interface{})
+				child := findByName(parent.Children, path)
+				if child == nil {
+					panic(fmt.Errorf("could not find child '%s' for path '%s' in parent '%s'", path, f.Name, parent.Name))
```

Comment:
If this really should be fatal, please use `model.fatal`. However I am not sure this should be fatal: In theory a remote might send us index updates in an ugly order, where a child comes in a batch before the parent. We could just skip it.

Reply:
So this is used in the http handler, so it would just return a 500 etc. But you're right I guess (value, err) is a better approach here.